src/odemis/driver/zeiss.py

```diff
@@ -0,0 +1,963 @@
+                     self._hfw_nomag / MAGNIFICATION_RANGE[0])
+        self.horizontalFoV = model.FloatContinuous(self._hfw_nomag / self.magnification.value,
+                                                   range=fov_range, unit="m",
+                                                   setter=self._setHorizontalFoV)
+        self.pixelSize = model.VigilantAttribute(self.parent.GetPixelSize(),
+                                                 unit="m", readonly=True)
+        self.blanker = model.VAEnumerated(self.parent.GetBlankBeam(), choices={True, False},
+                                          setter=self._setBlanker)
+        self.external = model.VAEnumerated(self.parent.GetExternal(), choices={True, False},
+                                          setter=self._setExternal)
+
+        # Refresh regularly the values, from the hardware, starting from now
+        self._updateSettings()
+        self._va_poll = util.RepeatingTimer(5, self._updateSettings, "Settings polling")
+        self._va_poll.start()
+
+    def _updateSettings(self):
+        """
+        Read all the current settings from the SEM and reflects them on the VAs
+        """
+
+        logging.debug("Updating SEM settings")
+        try:
+            mag = self.parent.GetMagnification()
+            if mag != self.magnification.value:
+                # Update both horizontalFoV, and magnification
+                if MAGNIFICATION_RANGE[0] <= mag <= MAGNIFICATION_RANGE[1]:
+                    self.magnification._set_value(mag, force_write=True)
+                    fov = self._hfw_nomag / mag
+                    self.horizontalFoV._value = fov
+                    self.horizontalFoV.notify(fov)
+                else:
+                    logging.warning("Hardware reports magnification = %g, outside of expected range", mag)
+
+            blanked = self.parent.GetBlankBeam()
+            if blanked != self.blanker.value:
+                self.blanker._value = blanked
+                self.blanker.notify(blanked)
+            external = self.parent.GetExternal()
+            if external != self.external.value:
+                self.external._value = external
+                self.external.notify(external)
+            pixelSize = self.parent.GetPixelSize()
+            if pixelSize != self.pixelSize.value:
+                self.pixelSize._value = pixelSize
+                self.pixelSize.notify(pixelSize)
+
+        except Exception:
+            logging.exception("Unexpected failure when polling settings")
+
+    def _setExternal(self, scanctrl):
+        self.parent.external_scanning(scanctrl)
+        return scanctrl
+
+    def _setHorizontalFoV(self, fov):
+        self.parent.SetFoV(fov)
+        return fov
+
+    def _setBlanker(self, blankctrl):
+        self.parent.SetBlankBeam(blankctrl)
+        return bool(blankctrl)
+
+    def _onHorinzontalFoV(self, fov):
+        # => pixelSize + magnification
+        # TODO:
+        pass
```

Comment:
I'm a bit confused about that one. What do you want me to implement? I thought the zeiss sem doesn't have a fov attribute, but only a magnification from which the fov can be calculated.